src/JustEat.StatsD/SimpleObjectPool.cs

```diff
@@ -8,7 +8,7 @@ namespace JustEat.StatsD
     public sealed class SimpleObjectPool<T>
         where T : class
     {
-        private readonly Stack<T> _pool;
+        private readonly ConcurrentStack<T> _pool;
```

Comment:
This is better. Although I'm not sure why it was a stack in the first place. I would have thought a ```ConcurrentBag``` would be a better fit.
If you look on MSDN for object pooling, ```ConcurrentBag``` is what they use:
https://msdn.microsoft.com/en-us/library/ff458671(v=vs.110).aspx